src/common/file_stream.h

```diff
@@ -205,12 +250,27 @@ class OutputFileStream {
   template <typename T>
   friend OutputFileStream& operator<<(OutputFileStream& stream, const T& t) {
     stream.ostream_ << t;
+    ABORT_IF(!stream.ostream_,
```

Comment:
Here we are checking for both a badbit and failbit as `!` is a synonym of `fail()`. Above (in `getline` and read operations) only a badbit is checked with `bad()`. This should be unified. Or is this some difference between Input and OutputFileStream?

Reply:
I think you cannot do that for InputFileStream in the same way. Consider:
```
while(io::getline(stream, line)) {
   …
}
```

This is also a Boolean check on steam and stops when EOF is reached. I think there is no equivalent for OutputFileStream.

Reply:
My question was whether we should use `fail()` instead of `bad()`, not about boolean check.

Reply:
Where in the code is that?

Reply:
OK found it. Rather comment in the code where the error is, not where it is not :)
It seems that checking for fail() checks both, failbit and badbit, so we should indeed use that.

Reply:
Yes, `fail()` does check for both. OK, I'll change that.

Reply:
Reaching EOF seems to set failbit. So this cannot be used for std::getline, I guess?